File: 2394-count-subarrays-with-score-less-than-k/count-subarrays-with-score-less-than-k.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        n = len(nums)
        left = 0
        sum_ = 0
        count = 0
        
        for right in range(n):
            sum_ += nums[right]
            
            while left <= right and sum_ * (right - left + 1) >= k:
                sum_ -= nums[left]
                left += 1
                
            count += right - left + 1
        
        return count

File: 2394-count-subarrays-with-score-less-than-k/count-subarrays-with-score-less-than-k.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, long long k) {
        long long n = nums.size();
        long long left = 0, right = 0; 
        long long sum = 0, count = 0; 
        
        while(left < n ){
            //Find the larget valid window 
            while (right < n and (sum+nums[right])*(right-left+1)<k)
                sum += nums[right++];
            count += right-left; 

            //slide the window forward by removing nums[left]
            if (right == left)
                right++;
            else
                sum -= nums[left];
            
            left++;
        }
        return count;
        
    }
};